server/auth_server.go

```diff
@@ -370,6 +370,22 @@ func updateSettings(saveConfig map[string]map[string]string) error {
 		// authConfig now was prevConfig
 		// saveConfig is to be saved, so authConfig should get added values from saveConfig
 		log.Debugf("Previous saved auth config: %v", authConfig)
+		// saveConfig (updated config) might contain no secret settings, if GetConfig(listOnly=true) was called before it
```

Comment:
This wasn't a problem earlier because the filter on listOnly wasn't added until the bug for serviceAccountPassword being pre-entered in UI was logged. So GetConfig was always returning all values, and updateSettings would always get the secret values